File: POO/aula16/JAgencia/ContaInvestimento.java
class ContaInvestimento extends Conta
{
	private float taxaRendimento;
	private float taxaAdministracao;

	public ContaInvestimento()
	{	}

	public ContaInvestimento(String nome, String cpf, float taxa)
	{
		super(nome, cpf);
		this.setTaxaRendimento(taxa);
		this.setTaxaAdministracao(0.0f);
	}

	public void setTaxaRendimento(float taxa)
	{
		this.taxaRendimento = taxa;
	}

	public void setTaxaAdministracao(float taxa)
	{
		this.taxaAdministracao = taxa;
	}

	public float getTaxaRendimento()
	{
		return this.taxaRendimento;
	}

	public float getTaxaAdministracao()
	{
		return this.taxaAdministracao;
	}

	public void correcao()
	{
		this.setSaldo(this.getSaldo() * (1.0f + (this.getTaxaRendimento()/100.0f)));
	}

	public void extrato()
	{
		super.extrato();
		System.out.println("TAXA RENDIMENTO    : " + this.getTaxaRendimento() + "/%a.m.");
		System.out.println("TAXA ADMINISTRACAO : " + this.getTaxaAdministracao() + "/%a.m.");
		System.out.println("============================");
	}
}

File: POO/aula16/ProjetoAgencia/ContaInvestimento.cpp
#include "ContaInvestimento.h"

ContaInvestimento::ContaInvestimento(std::string nome, std::string cpf, float taxa) : Conta(nome, cpf)
{
	this->setTaxaRendimento(taxa);
	this->setTaxaAdministracao(0);
}

void ContaInvestimento::setTaxaRendimento(float taxa)
{
	this->taxaRendimento = taxa;
}

void ContaInvestimento::setTaxaAdministracao(float taxa)
{
	this->taxaAdministracao = taxa;
}

float ContaInvestimento::getTaxaRendimento()
{
	return this->taxaRendimento;
}

float ContaInvestimento::getTaxaAdministracao()
{
	return this->taxaAdministracao;
}

void ContaInvestimento::correcao()
{
	this->setSaldo(this->getSaldo() * (1.0f+(this->getTaxaRendimento()/100.0f)));
}

void ContaInvestimento::extrato()
{
	Conta::extrato();
	std::cout << "TAXA DE RENDIMENTO: " << this->getTaxaRendimento() << "\%a.m." << std::endl;
	std::cout << "TAXA DE ADMINISTRACAO: " << this->getTaxaAdministracao() << "\%a.m." << std::endl;
	std::cout << "===========================================" << std::endl;
}
